### DISCUSSION TITLE
High CPU usage compared to `watch` equivalent

### DISCUSSION THREAD

**Discussion by xfzv** [sachaos/viddy#141] [Q&A] [answered]
Not sure whether this behaves correctly or doesn't so I thought a discussion would be more appropriate for now.

I've been playing around with [v1.0.0](https://github.com/sachaos/viddy/releases/tag/v1.0.0) and [v1.0.1](https://github.com/sachaos/viddy/releases/tag/v1.0.1) and on a server with limited resources (2 cores, 4 GB of RAM).

```console
% watch -n 10 `my command`
```

![watch](https://github.com/user-attachments/assets/07ec10b0-1a21-4394-83d0-3b7261f95860)

---

```console
% viddy -n 10 `my command`
```

![viddy](https://github.com/user-attachments/assets/53b14305-5a3e-4577-a7a9-bf441df586fc)

The CPU consumption is constant for both commands. I tried other commands and it seems that the more output lines the command produces, the higher the CPU usage.

With another command that produces less output:
```console
% watch -n 10 `my command`
```
![watch](https://github.com/user-attachments/assets/d8a349c8-dc36-4faf-815e-e3b390e4cb4d)

---

```console
% viddy -n 10 `my command`
```

![viddy](https://github.com/user-attachments/assets/8165443e-6446-4f01-94c8-d839fb7b105d)

Note that `watch` consumes the same amount of CPU/RAM for both commands.

`viddy` CPU usage for the first command is really high, so much that I'm sticking with `watch` for now.

Is anyone observing similar resource usage? Just curious.

**Comment by sachaos** [comment, marked as answer]
@xfzv Thank you for the feedback.
Yes, Viddy is heavier than Watch because it offers rich features, especially the pager functionality. However, there is potential for optimizing this feature in the future.

**Reply by xfzv** [reply]
I see. Thanks for the reply!
